hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemViewLock.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.view;
+
+import java.io.Serializable;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class HoodieTableFileSystemViewLock  implements Serializable {
+
+  // Locks to control concurrency. Sync operations use write-lock blocking all fetch operations.
+  // For the common-case, we allow concurrent read of single or multiple partitions
+  private final ReentrantReadWriteLock globalLock = new ReentrantReadWriteLock();
+  private final ReentrantReadWriteLock.ReadLock readLock = globalLock.readLock();
+  private final ReentrantReadWriteLock.WriteLock writeLock = globalLock.writeLock();
+
+  /**
+   * Acquire read lock
+   */
+  public void readLock() {
+    readLock.lock();
+  }
+
+  /**
+   * Release read lock
+   */
+  public void readUnlock() {
+    readLock.unlock();
+  }
+
+  /**
+   * Acquire write lock
+   */
+  public void writeLock() {
+    writeLock.lock();
+  }
+
+  /**
+   * Release write lock
+   */
+  public void writeUnlock() {
+    writeLock.unlock();
+  }
+
+  /**
+   * Check if the current thread holds write lock
+   */
+  public boolean hasWriteLock() {
+    return globalLock.isWriteLockedByCurrentThread();
```

Comment:
It seems like the return value is not used,  will it block when calling this method? Otherwise we don't need to call it.

Reply:
This is a reentrant read-write lock， and with this function we just confirm whether the lock has been acquired. The outer layer has actually acquired the writelock， there we just verify it.
![image](https://github.com/apache/hudi/assets/34104400/ac296a51-a7fb-4142-a409-3f9075692e47)

Reply:
How do we verify this? I mean,  do we need to check the return value of  `hasWriteLock`?

Reply:
> How do we verify this? I mean, do we need to check the return value of `hasWriteLock`?

I'll add a unit test later for this.

Reply:
> How do we verify this? I mean, do we need to check the return value of `hasWriteLock`?

Sorry, I've been busy lately. Added ut to check the value of hasWriteLock, cc~ @stream2000